core/src/main/java/com/novoda/merlin/MerlinsBeard.java

```diff
@@ -69,4 +69,12 @@ public boolean isConnectedToMobileNetwork() {
         return networkInfo != null && networkInfo.isConnected();
     }
 
+    public String getMobileNetworkSubtypeName() {
+        NetworkInfo networkInfo = getNetworkInfo();
+        if (networkInfo == null || !networkInfo.isConnected()) {
+            return "";
```

Comment:
Yep I meant the return value, from an api standpoint it would be strange to
get nothing, but it would then also be a constant that you could check
against
On 5 Oct 2015 09:11, "Ataul Munim" <notifications@github.com> wrote:

> In core/src/main/java/com/novoda/merlin/MerlinsBeard.java
> <https://github.com/novoda/merlin/pull/66#discussion_r41119052>:
>
> > @@ -69,4 +69,12 @@ public boolean isConnectedToMobileNetwork() {
> >          return networkInfo != null && networkInfo.isConnected();
> >      }
> >
> > +    public String getMobileNetworkSubtypeName() {
> > +        NetworkInfo networkInfo = getNetworkInfo();
> > +        if (networkInfo == null || !networkInfo.isConnected()) {
> > +            return "";
>
> [image: :+1:] `return "unknown"; is also good for me. But not sure that's
> what he meant [image: :tongue:]
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/novoda/merlin/pull/66/files#r41119052>.
>